runtime/lua/man.lua

```diff
@@ -162,7 +162,7 @@ local function highlight_man_page()
   end
   buf_hls = {}
 
-  vim.api.nvim_command("let &modifiable = "..mod)
```

Comment:
This is not what actually should be done: there is a specific API to get/set options: `vim.api.nvim_buf_get_option` and `vim.api.nvim_buf_set_option`. All `nvim_command` and `nvim_eval` occurrences in this function should be replaced with that, it is also more typesafe as it should be using booleans.

Also note that your fix is incorrect even without that API: lua documentation does not state how it will represent numbers when coercing to integers, so you must not use concat and floor and must use `format` method of strings instead.

Reply:
@ZyX-I thanks for the quick reply.  I do not have much experience with lua.

Reply:
By the way, I do not see how `mod` may receive any value, but 0 and 1 here, so you may have hit problem like this one: see http://www.lua.org/manual/5.3/manual.html#8.1, it states

> The conversion of a float to a string now adds a .0 suffix to the result if it looks like an integer. (For instance, the float 2.0 will be printed as 2.0, not as 2.) You should always use an explicit format when you need a specific format for numbers.

Reply:
Note though that if this is the case then nobody have tested our lua code against lua 5.3, due to incompatibilities many distributions still use 5.1 and do not plan on updating. For example, Gentoo has introduced slotting (slotting allows installing multiple incompatible versions of some slotted package) to lua ebuild, and new slotted luas are still masked as experimental while not slotted lua is still 5.1.5.

Reply:
I use Fedora 27, we only have Lua 5.3 (hence probably this problem).

Reply:
> Note though that if this is the case then nobody have tested our lua code against lua 5.3

LuaJIT is specifically compatible with Lua 5.1, so we're tired to that version of Lua.